parcel-bundler/parcel
# Add parcel-link and parcel-unlink dev CLIs

**NOTE**: This PR is stacked on #8857 because its tests rely on changes from that PR

This is a packaging of a very useful script that @mischnic initially created for our internal use. The script makes it easy to test a local copy of Parcel in an existing project by replacing Parcel packages with symlinks to the local copy.

In this PR, the script has been wrapped in a CLI, such that it can be installed and used to link any copy of Parcel to any project.

## How to test
check out this branch and run `yarn`, then `cd packages/dev/parcel-link && yarn link`
to make the `parcel-link` binary globally available.

Then, you should be able to try linking into any project by `cd /project/path && parcel-link`.

You can also link a different copy of Parcel (default is the copy that `parcel-link` belongs to:
```sh
$ cd /project/path && parcel-link /path/to/parcel/packages
```

The README has more info, as does `parcel-link -h`

## unlinking

Currently, `parcel-link unlink` attempts to do the opposite of what `parcel-link` did, as recorded in the `.parcel-link` manifest.

The idea is that `parcel-link unlink` should "just work", rather than taking options to match `parcel-link`.

## TODO
- [x] make parcel-unlink work without arguments
- [x] test in a variety of projects
